src/main/webapp/go.js

```diff
@@ -426,7 +434,54 @@ function checkUserDestinationGuess() { //eslint-disable-line
     return;
   }
 
-  // If userGuess is not an exact match, perform entity extraction.
+  // Check to see if userGuess can be used to identify the correct place
+  // using the Places library.
+  checkGuessWithPlaceIDs(userGuess);
+}
+
+/**
+ * Retrieves place ID for the answer, and compares this to the place
+ * ID retrieved for the user's guess.
+ * @param {String} userGuess User's guess for current destination.
+ */
+function checkGuessWithPlaceIDs(userGuess) {
+  const answerRequest = createRequestForPlaceID(hunt.getCurDestName());
+  service.findPlaceFromQuery(answerRequest, function(results, status) {
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      const answerID = results[0].place_id;
+      getPlaceIDOfGuess(userGuess, answerID);
+    }
+  });
+}
+
+/**
+ * @param {String} userGuess The user's guess for the name of the
+ * current destination.
+ * @param {String} answerID Place ID corresponding to the correct
+ * destination
+ */
+function getPlaceIDOfGuess(userGuess, answerID) {
```

Comment:
This method isn't just getting the Place ID of the guess.  If it doesn't match it falls back to entity extraction.